osquery/core/database/rocksdb_database.cpp

```diff
@@ -0,0 +1,388 @@
+/**
+ *  Copyright (c) 2018-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#ifdef OSQUERY_WINDOWS
+#include <Winsock2.h>
+#else
+#include <arpa/inet.h>
+#endif
+
+#include <osquery/core/database/rocksdb_database.h>
+#include <osquery/logger.h>
+
+#include <boost/algorithm/string.hpp>
+
+namespace osquery {
+
+/// Hidden flags created for internal stress testing.
+HIDDEN_FLAG(int32, rocksdb_write_buffer, 16, "Max write buffer number");
+HIDDEN_FLAG(int32, rocksdb_merge_number, 4, "Min write buffer number to merge");
+HIDDEN_FLAG(int32, rocksdb_background_flushes, 4, "Max background flushes");
+HIDDEN_FLAG(uint64, rocksdb_buffer_blocks, 256, "Write buffer blocks (4k)");
+
+static const int kMaxLogMessageBufferSize = 65536;
+
+static const int8_t kIntTag = 0x1;
```

Comment:
Something a bit more random/magic would feel safer.

Reply:
0x1 one of the safest options, if string is int size and contains 0x1 we will have much bigger problems in other parts of osquery.

Reply:
May be something above ASCII table, like `0x81` or similar.

Reply:
0x81 is not control char, so I don't think that it will be safer that 0x1

Reply:
I'm not sure that 0x01 can lead to any problems at all. `x01` is valid character for the C++ and C string functions. The only problematic character for all of them is `\0`.
But if string starts with `0x81` (we are using utf-8, right?) it is broken utf-8 string.